x-pack/plugins/lists/public/exceptions/components/builder/exception_items_renderer.tsx

```diff
@@ -395,11 +399,14 @@ export const ExceptionBuilderComponent = ({
   }, [exceptions, handleAddNewExceptionItem]);
 
   useEffect(() => {
-    if (exceptionListItems.length > 0) {
+    if (
+      exceptionListItems.length > 0 &&
+      !deepEqual(exceptionListItems, currentExceptionListItems)
+    ) {
+      setCurrentExceptionListItems(exceptionListItems);
       setUpdateExceptions(exceptionListItems);
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [currentExceptionListItems, exceptionListItems, setUpdateExceptions]);
```

Comment:
Noticed an issue, when added new exception item and then typed name.
Exception item has disappeared. I suspect it's because it got reset when name changed (probably because `exceptionListItems` has another reference)

https://github.com/elastic/kibana/assets/92328789/d810d543-986e-4f95-84a0-dbb87990d58d